test/algconvergence/partitioned_methods_tests.jl

```diff
@@ -368,6 +378,76 @@ sol = solve(prob, ERKN5(), reltol = 1e-8)
 sol = solve(prob, ERKN7(), reltol = 1e-8)
 @test length(sol.u) < 38
 
+# Testing generalized Runge-Kutte-Nyström methods on velocity dependend ODEs with the damped oscillator
+println("In Place")
+
+# Damped oscillator
+prob = ODEProblem(DynamicalODEFunction{false}((du, u, p, t) -> -u - 0.5 * du,
+        (du, u, p, t) -> du,
+        analytic = (du0_u0, p, t) -> OrdinaryDiffEq.SciMLBase.ArrayPartition([
+                exp(-t / 4) / 15 * (15 * du0_u0[1] * cos(sqrt(15) * t / 4) -
+                 sqrt(15) * (du0_u0[1] + 4 * du0_u0[2]) * sin(sqrt(15) * t / 4)),
+            ], # du
+            [
+                exp(-t / 4) / 15 * (15 * du0_u0[2] * cos(sqrt(15) * t / 4) +
+                 sqrt(15) * (4 * du0_u0[1] + du0_u0[2]) * sin(sqrt(15) * t / 4)),
+            ])),
+    OrdinaryDiffEq.SciMLBase.ArrayPartition([0.0], [1.0]), # du0, u0
+    (0.0, 10.0), # tspan
+    DiffEqBase.NullParameters(), # p
+    SecondOrderODEProblem{false}())
```

Comment:
@ChrisRackauckas: Is there a less ugly looking way to create this - ideally using the `SecondOrderODEProblem` constructor? This is just a working version we came up with.

Reply:
Why would this not use `SecondOrderODEProblem`? I'm a bit confused as to how this would be different.

Reply:
I couldn't find a constructor of a `SecondOrderODEProblem` that accepts an analytic solution. The only way I found to make it work was to write the conversion to an `ODEProblem` with `DynamicalODEFunction` and `ArrayPartition` by hand. That's okay and works well for the tests - I was just wondering whether there is a less manual way to achieve this.

Reply:
I think this shouldn't block this PR from being merged. I see some reasons why it would be difficult to get another nice interface with the current structure right now. Thus, we may file an issue to remember this but should go ahead with this PR.